[esx_addons]/esx_basicneeds/client/main.lua

```diff
@@ -74,38 +71,18 @@ AddEventHandler('esx_basicneeds:isEating', function(cb)
 	cb(IsAnimated)
 end)
 
-RegisterNetEvent('esx_basicneeds:onEat')
-AddEventHandler('esx_basicneeds:onEat', function(prop_name)
-	if not IsAnimated then
-		prop_name = prop_name or 'prop_cs_burger_01'
-		IsAnimated = true
-
-		CreateThread(function()
-			local playerPed = PlayerPedId()
-			local x,y,z = table.unpack(GetEntityCoords(playerPed))
-			local prop = CreateObject(joaat(prop_name), x, y, z + 0.2, true, true, true)
-			local boneIndex = GetPedBoneIndex(playerPed, 18905)
-			AttachEntityToEntity(prop, playerPed, boneIndex, 0.12, 0.028, 0.001, 10.0, 175.0, 0.0, true, true, false, true, 1, true)
-
-			ESX.Streaming.RequestAnimDict('mp_player_inteat@burger', function()
-				TaskPlayAnim(playerPed, 'mp_player_inteat@burger', 'mp_player_int_eat_burger_fp', 8.0, -8, -1, 49, 0, 0, 0, 0)
-				RemoveAnimDict('mp_player_inteat@burger')
-
-				Wait(3000)
-				IsAnimated = false
-				ClearPedSecondaryTask(playerPed)
-				DeleteObject(prop)
-			end)
-		end)
-
-	end
-end)
-
-RegisterNetEvent('esx_basicneeds:onDrink')
-AddEventHandler('esx_basicneeds:onDrink', function(prop_name)
+RegisterNetEvent('esx_basicneeds:onUse')
+AddEventHandler('esx_basicneeds:onUse', function(type, prop_name)
```

Comment:
There should be some way to make the code backwards compatible. If somebody uses this esx_basicneeds:onEat event in another script, the code should not break. 

Do you think there are people who use this event in other scripts? 

A possible solution could be to throw some message that it has changed and make the changes.

What do you think about these?

Reply:
I think this is now acceptable. Thanks for the correction.